api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/Eri2MonthsCohortQueries.java

```diff
@@ -0,0 +1,338 @@
+    cd.addSearch(
+        "pregnant",
+        EptsReportUtils.map(
+            getPregnantWomenRetainedOnArtFor2MonthsFromArtInitiation(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "breastfeeding",
+        EptsReportUtils.map(
+            getBreastfeedingWomenRetainedOnArtFor2MonthsFromArtInitiation(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+    cd.setCompositionString("(all AND adults) AND NOT (pregnant OR breastfeeding)");
+    return cd;
+  }
+
+  /**
+   * Get patients who did not pick up drugs on their second visit Did NOT pick up the drugs A and B
+   *
+   * @return CohortDefinition
+   */
+  public CohortDefinition getPatientsWhoDidNotPickDrugsOnTheirSecondVisit() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+    cd.setName("Patients who did not pick up drugs during their second visit");
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+    cd.addSearch(
+        "initiatedArt",
+        EptsReportUtils.map(
+            getAllPatientsWhoInitiatedArt(), "endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "pickedDrugs",
+        EptsReportUtils.map(
+            getAllPatientsWhoReturnedFor2ndConsultationOR2ndDrugsPickUpWithin33Days(),
+            "endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "dead",
+        EptsReportUtils.map(
+            genericCohortQueries.getPatientsBasedOnPatientStates(
+                hivMetadata.getARTProgram().getProgramId(),
+                hivMetadata.getPatientHasDiedWorkflowState().getProgramWorkflowStateId()),
+            "endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "transfers",
+        EptsReportUtils.map(
+            genericCohortQueries.getPatientsBasedOnPatientStates(
+                hivMetadata.getARTProgram().getProgramId(),
+                hivMetadata
+                    .getTransferredOutToAnotherHealthFacilityWorkflowState()
+                    .getProgramWorkflowStateId()),
+            "endDate=${endDate},location=${location}"));
+    cd.setCompositionString("initiatedArt AND NOT (pickedDrugs OR dead OR transfers)");
+    return cd;
+  }
+
+  /**
+   * Get patients who picked up drugs on their second visit A and B and C and NOt (dead and
+   * tarnsfers)
+   *
+   * @return CohortDefinition
+   */
+  public CohortDefinition getPatientsWhoPickedUpDrugsOnTheirSecondVisit() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+    cd.setName("Patients who  picked up drugs during their second visit");
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+    cd.addSearch(
+        "pickedDrugsAndStartedART",
```

Comment:
I think there might be a problem here.

`getPatientsWhoDidNotPickDrugsOnTheirSecondVisit` = `A and B and not (C or DEAD or TRANS)`

Which means that this cohort is `(A and B and not (C or DEAD or TRANS)) and not (dead OR trans)`
but in the user story this should be: `A and B and C and Not (Dead or Transferred out)`